components/smalltalk/squeak5/Makefile

```diff
@@ -0,0 +1,80 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Package Makefile for the "OpenSmalltalk Cog Spur" Squeak Smalltalk V5 system
+# See http://squeak.org and http://opensmalltalk.org
+#
+
+BUILD_BITS=64
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak5
+COMPONENT_VERSION=	5.0.6
+GIT_TAG=		sun-v$(COMPONENT_VERSION)
+PLUGIN_REV=		5.0-202005010835-sun
+COMPONENT_SUMMARY=	The Squeak V5 Smalltalk Virtual Machine
+COMPONENT_PROJECT_URL=	http://www.squeak.org
+COMPONENT_FMRI=		runtime/squeak5
+COMPONENT_CLASSIFICATION=	Development/Other Languages
```

Comment:
For me it's ok that you have introduced the new category "smalltalk" in the file system hierarchy. We need to wait for alp's decission whether the new package should be moved to the runtime folder. But I would like to have the COMPONENT_FRMI match the category (so in the end the squeak5 folder should either be moved into the runtime folder or the COMPONENT_FRMI should be changed to smalltalk/squeak5.

Remember, when everything is done you should squash all commits into a single one by means of git rebase.

Reply:
The fmri runtime/squeak5 was chosen randomly without much thought.

I like the fmri smalltalk/squeak5 better.

Also I have been thinking about just calling the package "smalltalk/squeak" but the problem that I have there, is that the "build" system of squeak4 and squeak5 is different and that perhaps a mediator package could be made called smalltalk/squeak.

I'll change it to use smalltalk/squeak5 but if I understand correctly the decision in the end depends on the folder that will be used in the end ...

Perhaps it makes sense to have a folder 'smalltalk' given the number of packages that could be stored there, given the number of different implementations and dialects.

Reply:
You cannot randomly choose it. Eg. the file hierarchy is documented in doc/categories-layout.md

Reply:
Please, don't introduce new top-level FMRI pkg:/smalltalk/blah. 
A new component's category is justified if there's a dozen sub-components (e.g, if you plan to add several smalltalk libraries). If these are just compilers, they can live in developer/ (with corresponding FMRI) for compilers and  un runtime/ (with corresponding FMRI) for interpreters.

Reply:
Sorry, so far I don't have time to look at this PR.

Reply:
In my judgement the Squeak VM is an "interpreter" although https://squeak.org/features/ states (probably quite rightfully so) that it provides "just-in-time compilation, stack-to-register mapping, and aggressive in-line message caching yield efficiency in executing Smalltalk byte code".     But basically given pyhalov's remark to either use developer/ or runtime/  I think runtime/ was the right choice.   This is what I chose initially.   Then AndWac's remark is just that I have to move the component to the components/runtime/ directory and get rid of the components/smalltalk directory.   I thought that adding a components/smalltalk directory made sense, but it is also true that this is not compatible with the "fmri" name.

Reply:
I am going to close the squeak4 and squeak5 pull requests.   And submit a new pull request called "squeak" containing the 32-bit squeak4 and 64-bit squeak5 packages in one pull request: called "Add squeak".    I'll add the packages to the components/runtime folder.   Also in the future I will modify the squeak5 package to support both 32-bit and 64-bit , and I can also modify squeak4 to support both 32-bit and 64-bit, but currently squeak4 is 32-bit only and squeak5 is 64-bit only.   So I will do a new pull request now for squeak5 version 5.0.7 (going up one level and this will also incorporate new changes from opensmalltalk-vm).

Reply:
That is ok, but you should be aware that you'll open an new can of worms. Eg. mediator is not a separate package but part of every package that is an alternative. See for an example the different node packages in the runtime folder. So if you put everything in one package you cannot use the mediator feature to let users choose the default Smalltalk they want to use. They always need to add the corresponding path.

Reply:
I am for the moment moving the packages to components/runtime.    Smalltalk is like "ruby" and "python" in the sense that Smalltalk may provide lots of "modules" (interpreted by the "vm" or virtual machine).    But I'm not convinced that a folder "smalltalk" is required.   We could change it later again.   The main point of "AndWac"'s remark is that component folder and FMRI in the package name should match, if I understand correctly.

Reply:
Not exactly. We have some packages which differ but for new packages I prefer it (easier to find the package sources).